packages/terra-show-hide/src/ShowHide.jsx

```diff
@@ -44,6 +51,42 @@ const defaultProps = {
   buttonAlign: 'start',
   isOpen: false,
   preview: undefined,
+  focusRef: null,
+};
+
+/**
+ * Disables focus styles for normally unfocusable elements.
+ */
+const disableFocusStyles = (event) => {
+  event.currentTarget.setAttribute('data-focus-styles-enabled', 'false');
+};
+
+/**
+ * Remove attributes that makes element focusable.
+ */
+const removeAttributes = (event) => {
+  event.currentTarget.removeAttribute('tabindex');
+  event.currentTarget.removeAttribute('role');
+};
+
+/**
+ * Focus element.
+ */
+const focusElement = (element) => {
+  element.setAttribute('data-focus-styles-enabled', 'true');
+  element.addEventListener('blur', disableFocusStyles);
```

Comment:
Do we really need this logic to set the attribute?  We should be able to create CSS such that any descendant of the component that has focus gets the new style.

Reply:
This is a good question. I noticed, that some of terra components use styling on focus like this: `.button[data-focus-styles-enabled='true']:focus`, so I added that attribute for consistency, in order to not loose focus styling for these components. Let me do some more research as for how many components actually use that approach and get back with the update on focus styling approach.

Reply:
I double checked and I think it's safe to remove it as the usage of that attr was in terra-docs button, so I did: https://github.com/cerner/terra-core/pull/3838/commits/b87f7ad08f6658a55f44d86bab0143406f6ccae1